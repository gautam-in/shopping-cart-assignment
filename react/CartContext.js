import React from 'react';

// set the defaults
const CartContext = React.createContext({
  cart: [],
  setCart: () => {},
});

export default CartContext;
