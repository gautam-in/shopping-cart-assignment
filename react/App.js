import React, { useState } from 'react';
import Home from './Components/home';
import Products from './Components/products';
import Login from './Components/login';
import Register from './Components/register';
import Nav from './Components/Nav';
import './Styles/main.css';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import CartContext from './CartContext';

export default function () {
  const [cart, setCart] = useState([]);
  const value = { cart, setCart };

  return (
    <CartContext.Provider value={value}>
      <Router>
        <header className="header">
          <Nav />
        </header>
        <main>
          <Switch>
            <Route path="/login">
              <Login />
            </Route>
            <Route path="/register">
              <Register />
            </Route>
            <Route path="/products/:categoryId">
              <Products />
            </Route>
            <Route path="/products">
              <Products />
            </Route>
            <Route path="/">
              <Home />
            </Route>
          </Switch>
        </main>
      </Router>
    </CartContext.Provider>
  );
}
