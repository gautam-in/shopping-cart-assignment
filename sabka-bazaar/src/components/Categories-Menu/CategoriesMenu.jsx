import PropTypes from "prop-types";
import "./categories-menu.styles.scss";
import Select from "react-select";
import variables from "../../_variables.module.scss";
import { memo } from "react";
import SideBar from "./SideBar";
const customStyles = {
  control: (styles) => ({
    ...styles,
    backgroundColor: variables.primary_color,
    display: "grid",
    gridTemplateColumns: "18fr 1fr",
  }),
  option: (styles) => {
    return { ...styles, color: variables.white_color };
  },
  placeholder: () => ({
    color: variables.white_color,
  }),
  dropdownIndicator: () => ({
    color: variables.white_color,
  }),
  indicatorSeparator: () => ({
    display: "none",
  }),
  menuList: () => ({
    backgroundColor: variables.primary_color,
  }),
  singleValue: () => ({
    color: variables.white_color,
  }),
  valueContainer: () => ({
    display: "flex",
    paddingLeft: "20px",
    alignitems: "center",
  }),
};
const CategoriesMenu = ({ data, selectCategoryId, selectedCategoryId }) => {
  const options = data.map(({ name, id }) => {
    return { value: name, label: name, id: id };
  });

  const selectCategory = (selectedOption) => {
    selectCategoryId(selectedOption.id);
  };
  return (
    <>
      <SideBar
        data={data}
        selectCategoryId={selectCategoryId}
        selectedCategoryId={selectedCategoryId}
      />
      <Select
        placeholder="Select category"
        options={options}
        className="categories-menu-select-drop-down"
        styles={customStyles}
        onChange={selectCategory}
      />
    </>
  );
};
export default memo(CategoriesMenu);
CategoriesMenu.propTypes = {
  data: PropTypes.array,
  isLoading: PropTypes.bool,
  selectCategoryId: PropTypes.func,
  selectedCategoryId: PropTypes.string,
};
