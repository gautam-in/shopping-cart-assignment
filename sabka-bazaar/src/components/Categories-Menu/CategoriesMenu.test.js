import { render } from "@testing-library/react";
import CategoriesMenu from "./CategoriesMenu";
describe("CategoriesMenu component", () => {
  it("should match snapshot", () => {
    const mockProps = {
      data: [
        {
          name: "Beverages",
          key: "beverages",
          description:
            "Our beverage department will ensure your fridge is always fully stocked. Shop for soda, juice, beer and more. ",
          enabled: true,
          order: 3,
          imageUrl: "/static/images/category/beverages.webp",
          id: "5b675e5e5936635728f9fc30",
        },
        {
          name: "Bakery Cakes and Dairy",
          key: "bakery-cakes-dairy",
          description:
            "The best cupcakes, cookies, cakes, pies, cheesecakes, fresh bread, biscotti, muffins, bagels, fresh coffee, milk and more.",
          enabled: true,
          order: 2,
          imageUrl: "/static/images/category/bakery.webp",
          id: "5b6899123d1a866534f516de",
        },
        {
          name: "Beauty and Hygiene",
          key: "beauty-hygiene",
          description:
            "Buy beauty and personal care products online in India at best prices.",
          enabled: true,
          order: 4,
          imageUrl: "/static/images/category/beauty.webp",
          id: "5b68994e3d1a866534f516df",
        },
      ],
      isLoading: false,
      selectCategoryId: "",
    };
    const container = render(<CategoriesMenu data={mockProps.data} />);
    expect(container).toMatchSnapshot();
  });
});
