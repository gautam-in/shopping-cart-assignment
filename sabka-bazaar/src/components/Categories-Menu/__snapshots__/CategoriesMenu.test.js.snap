// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesMenu component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <aside
        class="categories-menu-container"
      >
        <div
          class="category-menu-item"
          role="button"
          tabindex="0"
        >
          Beverages
        </div>
        <div
          class="category-menu-item"
          role="button"
          tabindex="0"
        >
          Bakery Cakes and Dairy
        </div>
        <div
          class="category-menu-item"
          role="button"
          tabindex="0"
        >
          Beauty and Hygiene
        </div>
      </aside>
      <div
        class="categories-menu-select-drop-down css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-j5toe4-control"
        >
          <div
            class=" css-1knrnc3-ValueContainer"
          >
            <div
              class=" css-1njyful-Placeholder"
              id="react-select-2-placeholder"
            >
              Select category
            </div>
            <div
              class=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-2-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-2-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-1xjgjl1-IndicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-ksczov-DropdownIndicator"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <aside
      class="categories-menu-container"
    >
      <div
        class="category-menu-item"
        role="button"
        tabindex="0"
      >
        Beverages
      </div>
      <div
        class="category-menu-item"
        role="button"
        tabindex="0"
      >
        Bakery Cakes and Dairy
      </div>
      <div
        class="category-menu-item"
        role="button"
        tabindex="0"
      >
        Beauty and Hygiene
      </div>
    </aside>
    <div
      class="categories-menu-select-drop-down css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-j5toe4-control"
      >
        <div
          class=" css-1knrnc3-ValueContainer"
        >
          <div
            class=" css-1njyful-Placeholder"
            id="react-select-2-placeholder"
          >
            Select category
          </div>
          <div
            class=" css-6j8wv5-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-2-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-2-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            class=" css-1xjgjl1-IndicatorSeparator"
          />
          <div
            aria-hidden="true"
            class=" css-ksczov-DropdownIndicator"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
